.github/workflows/release.yml

```diff
@@ -0,0 +1,46 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - v*
+
+jobs:
+  binaries:
+    strategy:
+      matrix:
+        goos: [linux]
+        goarch: [amd64, arm64, ppc64le, s390x]
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - name: Set up Go
+      uses: actions/setup-go@v3
+
+    - name: Build and upload binary
+      env:
+        GOOS: ${{ matrix.goos }}
+        GOARCH: ${{ matrix.goarch }}
+        CGO_ENABLED: "0"
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        echo Generating version file
+        go generate ./cmd
+
+        echo Building for $GOOS $GOARCH
+        go build -trimpath -ldflags='-s -w' -o dist/build/pebble ./cmd/pebble
+
+        ARCHIVE_FILE=pebble_${GITHUB_REF_NAME}_${GOOS}_${GOARCH}.tar.gz
+        echo Creating archive $ARCHIVE_FILE
+        cp COPYING dist/build
+        cp README.md dist/build
+        tar -czf dist/$ARCHIVE_FILE -C dist/build .
+
+        echo Uploading $ARCHIVE_FILE to release $GITHUB_REF_NAME
+        gh release upload $GITHUB_REF_NAME dist/$ARCHIVE_FILE
```

Comment:
I'm not an export on `gh release upload`, but what does the resulting release look like here?

I quite liked what @SimonRichardson proposed in the `juju-qa-jenkins` where the release notes can be generated too? https://github.com/canonical/juju-release-jenkins/pull/67/files#diff-c759a587ce8b6c2c352ae5e2e5c4ca62cad308c186d9ed844b4d8d823d1eb59bR65

Reply:
Looking at your [fork](https://github.com/benhoyt/pebble/releases/tag/v0.2.1), I think using `gh release create` might allow us to set nicer looking titles etc. too

Reply:
Well, the idea is that this job will kick in *after* you manually create a GitHub release (which will create the tag). Then this job kicks off with that release's tag, so you want to `gh release upload` to the just-created, release, rather than `gh release create`. I'm old-school and like to nicely curate release notes rather than just "here's a log of all the commits" (which is easy to link to in the curated release notes if we want).

So we'll set the release title manually too (I just typed in that title you linked to manually).

Reply:
Aha that makes sense. Maybe we should add a section in the HACKING doc covering how to make a release, then? :)

Reply:
Good call -- done.